### PR TITLE
Added article "Cypress + Storybook. Keeping test scenario, data and c…

### PR DESCRIPTION
**Pull request by daedalius**
…omponent rendering in one place."

Added ref to article from README.md
Fixed numeration in README.md

### CODE REVIEW

File: sections/tools/cypress-and-storybook-exposing-component-from-story.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# Cypress + Storybook. Keeping test scenario, data and component rendering in one place.
+
+### One Paragraph Explainer
+
+Many of us have chosen Cypress as a tool to test components hosted via Storybook/Styleguidist/Docz. [@NoriSte's example](./cypress-and-storybook.md) suggests creating some Storybook Stories, put components there and expose important data to the global variable in order to have access to the test. The nice approach actually, but the test becomes broken into the pieces between Storybook and Cypress.
```
Comment (muratkeremozcan) [inline sections/tools/cypress-and-storybook-exposing-component-from-story.md:5]: "_This is a nice approach actually"_

File: sections/tools/cypress-and-storybook-exposing-component-from-story.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+Here I'd like to show how to go a little bit further and get the most out of executing JavaScript in Cypress. To see it in action, you may [download the source code](https://github.com/daedalius/article-exposing-component-from-storybook-to-handle-them-in-cypress) and execute then `npm i` and `npm test` in the console.
+
+- You may expose the component reference from Storybook Story to test it whatever you wish in Cypress (without breaking testing logic into pieces).
```
Comment (muratkeremozcan) [inline sections/tools/cypress-and-storybook-exposing-component-from-story.md:9]: _"to test it however you wish"_

File: sections/tools/cypress-and-storybook-exposing-component-from-story.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+### Should I choose the approach suggested?
+
+- If you use tests as a development environment - definitely, Yes!
+- If you look at tests as at **live** documentation - Yes.
+- If you really write unit-tests to cover things that too close to implementation and React-lifecycle - ... I don't know. I haven't been writing such a test for a long time. Are you sure that the covered logic is component responsibility? Maybe that logic should be extracted and tested accordingly?
+
+### Why not use cypress-react-unit-test then? Why do we need Storybook?
+
+Probably, it is our future to test components. There will be no need to maintain a separate instance of the Storybook, all tests will be entirely under the responsibility of Cypress, the configuration will be simplified, etc.
+But now the tool has [some](https://github.com/bahmutov/cypress-react-unit-test/issues/34) [problems](https://github.com/bahmutov/cypress-react-unit-test/issues/52) that make the environment provided incomplete for running tests. Hope that Gleb Bahmutov and the Cypress team will make it worked 🤞
```
Comment (muratkeremozcan) [inline sections/tools/cypress-and-storybook-exposing-component-from-story.md:219]: _"Cypress team will make it worked"_

**Comment by muratkeremozcan** [comment]
@daedalius  I REALLY like this, it's very cool. Sharing a simple working example is much appreciated, and it works well out of the box.

 Some tiny typos to fix, that's it. 

@NoriSte   I had a concern with the topic layout. Would you say _Component Testing_ is a best practice or _Doing Component Testing with Cypress & Storybook_? The 2 topics under 7 could fit in Tools, Advanced, or a Component Testing topic in my opinion. Also you want to cross reference Title 1 : Testing Strategies when mentioning component testing, because now you have a detailed section on how to do it.

File: sections/tools/cypress-and-storybook-exposing-component-from-story.md (hunk 3 above)
Comment (NoriSte) [reply sections/tools/cypress-and-storybook-exposing-component-from-story.md:219]: @muratkeremozcan what is the fix? Replacing the whole _Hope that Gleb Bahmutov and the Cypress team will make it worked 🤞_ sentence with _Cypress team will make it worked_?

**Comment by NoriSte** [comment]
@muratkeremozcan you are right, well:
- I created a new section because some more chapters are going to come in the next months, 
  - Component tests with react-unit-test (from me probably)
  - TestCafè + Storybook (from @makbeth)

- I'll the testing categorization chapter to consider this "new" Storybook + E2E test approach

**Comment by NoriSte** [comment]
One more chapter could come, take a look at [this tweet](https://twitter.com/GHengeveld/status/1245446157981159424)

**Comment by NoriSte** [comment]
@muratkeremozcan @daedalius what do you think about the latest changes? Are you ok to merge it?

**Comment by muratkeremozcan** [comment]
@NoriSte go for it

**Comment by daedalius** [comment]
+

**Comment by NoriSte** [comment]
Thanks guys 😊thank you @daedalius for your contribution!

**Pull request merged**